gradient-color-picker/GradientColorProvider.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Drawing;
+using System.Linq;
+
+namespace Messerli.GradientColorPicker
+{
+    public class GradientColorProvider : IGradientColorProvider
+    {
+        private readonly IReadOnlyCollection<GradientColorByValue> _gradientColorList;
+
+        public GradientColorProvider(IImmutableList<GradientColorByValue> gradientColorList)
+        {
+            _gradientColorList = gradientColorList.OrderBy(item => item.Value).ToImmutableList();
+        }
+
+        public Color PickColor(int value)
+        {
+            if (!_gradientColorList.Any())
+            {
+                throw new InvalidOperationException("No gradient color defined");
+            }
+
+            if (value <= _gradientColorList.First().Value)
+            {
+                return _gradientColorList.First().Color;
+            }
+
+            if (value >= _gradientColorList.Last().Value)
```

Comment:
Put `_gradientColorList.Last()` into a variable.

Reply:
Addressed in: https://github.com/messerli-informatik-ag/gradient-color-picker/pull/1/commits/be8d2e6f03d9373895cccf4f0b071f2c7166eb65